src/backend/commands/dbcommands.c

```diff
@@ -3357,7 +3359,8 @@ dbase_redo(XLogReaderState *record)
 		 */
 		{
 			XLogRecPtr	lsn = record->EndRecPtr;
-			SetLastWrittenLSNForDatabase(lsn);
+			if (set_lwlsn_db_hook)
+				set_lwlsn_db_hook(lsn);
```

Comment:
Remove the line above this and just pass record->EndRecPtr directly. Same thing in the hunk below.

Reply:
And then you can probably remote the extra block here too (the open bracket on 3360 and the close bracket on 3364)